client/API.md

```diff
@@ -0,0 +1,536 @@
+
+To get information about several tasks make a GET request to the `/task` endpoint.
+
+| Query Parameter | Purpose                                                                                                                                           |
+| --------------- | -------                                                                                                                                           |
+| pattern         | Filter results based on the pattern. Uses standard shell glob matching, see [this](https://golang.org/pkg/path/filepath/#Match) for more details. |
+
+#### Example
+
+Get all tasks.
+
+```
+GET /task
+```
+
+```
+{
+    "tasks" : [
+        {
+            "name" : "TASK_NAME",
+            "type" : "stream",
+            "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+            "enabled" : true,
+            "executing" : true,
+            "stats": {}
+        },
+        {
+            "name" : "ANOTHER_TASK_NAME",
+            "type" : "batch",
+            "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+            "enabled" : true,
+            "executing" : true,
+            "stats": {}
+        }
+    ]
+}
+```
+
+Optionally specify a glob `pattern` to list only matching tasks.
+
+```
+GET /task?pattern=TASK*
+```
+
+```
+{
+    "tasks" : [
+        {
+            "name" : "TASK_NAME",
+            "type" : "stream",
+            "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+            "enabled" : true,
+            "executing" : true,
+            "stats": {}
+        }
+    ]
+}
+```
+
+#### Response
+
+| Code | Meaning                |
+| ---- | -------                |
+| 200  | Success                |
+| 404  | No tasks match pattern |
+
+## Replays and Recordings
+
+Kapacitor can save recordings of data and replay them against a specified task.
+
+### Start Recording
+
+To create a recording make a POST request to the `/recording` endpoint.
+There are three methods for recording data with Kapacitor:
+
+| Method | Description                                        |
+| ------ | -----------                                        |
+| stream | Record the incoming stream of data.                |
+| batch  | Record the results of the queries in a batch task. |
+| query  | Record the result of an explicit query.            |
+
+All record POST requests must specify the parameter `method` with one of the above options.
+Each different `method` has it's own parameters.
+
+The request returns once the recording is started and does not wait for it to finish.
+A recording ID is returned to later identify the recording.
+
+##### Stream
+
+| Parameter | Purpose                                                             |
+| --------- | -------                                                             |
+| name      | Name of a task, used to only record data for the DBRPs of the task. |
+| duration  | Duration string indicating how long to record the stream.           |
```

Comment:
Is this a golang duration string?  It may be better to go with a rfc3339 date to indicate when to stop.  rfc3339 has much deeper support with other languages.